salt/state.py

```diff
@@ -149,6 +150,13 @@ def _gen_tag(low):
     return u'{0[state]}_|-{0[__id__]}_|-{0[name]}_|-{0[fun]}'.format(low)
 
 
+def _clean_tag(tag):
+    '''
+      urllib safe quote the tag value to avoid invalid chars in the filename
+    '''
+    return urllib.quote(tag.encode('utf8'), safe='')
```

Comment:
I don't think we want to force this to `UTF-8`. Could you instead use `__salt_system_encoding__`?

Reply:
Sounds good. If you can change this then we can re-review this. Thanks.